src/com/facebook/buck/swift/SwiftDescriptions.java

```diff
@@ -69,4 +60,27 @@ private SwiftDescriptions() {
     output.enableObjcInterop = Optional.of(true);
   }
 
+  static boolean hasSwiftSource(ImmutableSortedSet<SourceWithFlags> srcs) {
+    return Iterables.any(srcs, new Predicate<SourceWithFlags>() {
+      @Override
+      public boolean apply(SourceWithFlags input) {
+        SourcePath sourcePath = input.getSourcePath();
+        return (sourcePath instanceof PathSourcePath) &&
```

Comment:
@Coneko 
Since we couldn't obtain a proper instance of `SourcePathResolver` at this point (explained here https://github.com/facebook/buck/pull/847#discussion_r75711254) and my last attempt (by always add a swift companion target for the apple target) didn't work well because it messed up with the `TargetGraph`, we can't share the logic with `filterSwiftSources`. 
If we look at `SourcePathTypeCoercer`, there are only 2 possible types of a `SourcePath` in here, `BuildTargetSourcePath` and `PathSourcePath`, but `BuildTargetSourcePath` is not for element in `srcs` so we can cast sourcePath to `PathSourcePath` and check for its extension.